### ISSUE TITLE
HEAD001, HEAD002, HEAD003, BLOCK001, BLOCK002 and IMG001 never report anything through the CLI

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#57]
## Problem

Six rules never produce a finding when the linter runs through the CLI (`asciidoc-linter`) or `AsciiDocLinter.lint()`. Their unit tests pass, because those tests call `rule.check()` with plain strings. The CLI path hands them something else.

Affected rules: HEAD001, HEAD002, HEAD003, BLOCK001, BLOCK002, IMG001.

## Minimal reproduction

All at commit `52e7fa5982d980eefd5b548f68ff6f58767fe772`, run as `asciidoc-linter --format plain file.adoc`.

Skipped heading level (HEAD001 should fire):

```asciidoc
= T

== Eins

==== Sprung

text
```

Unterminated listing block (BLOCK001 should fire):

```asciidoc
= T

== Eins

----
code
```

Lowercase heading (HEAD002 should warn):

```asciidoc
= T

== eins klein

text
```

Missing block image (IMG001 should fire):

```asciidoc
= T

image::missing.png[]
```

## Expected vs. actual

- Expected: one finding per file (HEAD001, BLOCK001, HEAD002, IMG001).
- Actual: `✓ No issues found` and exit code 0 for all four files.

Rules that receive raw lines work fine. A Markdown heading (`## Markdown`, FMT004) or a Markdown table (FMT005) is reported as expected.

## Cause (confirmed)

`AsciiDocLinter.lint_string()` passes raw lines only to an explicit allow-list of rules. All other rules get the parser output:

- `asciidoc_linter/linter.py:104` — `document = self.parser.parse(content)`
- `asciidoc_linter/linter.py:110-123` — the `isinstance(rule, (WhitespaceRule, MarkdownSyntaxRule, ...))` branch; everything else runs `rule.check(document)`

`AsciiDocParser.parse()` (`asciidoc_linter/parser.py:43-55`) returns only `Header` dataclass objects, one per heading. The affected rules then skip every element because it is not a string:

- `asciidoc_linter/rules/heading_rules.py:77` (HEAD002), `:102` (HEAD001), `:169` (HEAD003) — `if isinstance(line, str)`
- `asciidoc_linter/rules/block_rules.py:78` (BLOCK001), `:159` (BLOCK002) — `if isinstance(line, str)`

IMG001 (`asciidoc_linter/rules/image_rules.py:136-140`) unwraps `.content`, but it only ever sees heading lines, so image macros in the body are never checked.

## Suggested fix

All rules already accept a list of strings. Pass `raw_lines` to every rule in `lint_string()`. Enabling the rules this way exposes two rule-level false positives, which should be fixed along with it:

- HEAD002 reports an example block delimiter (`====`) as "Missing space after ====".
- BLOCK002 reports every delimited block preceded by an attribute line (`[source,python]`) or a block title (`.Title`) as "Block should be preceded by a blank line".

A fix with regression tests that go through `AsciiDocLinter.lint()` follows as a PR.
